### ISSUE TITLE
Upgrading from v21 to v29 forces node pool recreation due to gcfs_config

### ISSUE THREAD

**Issue by davidgwps** [terraform-google-modules/terraform-google-kubernetes-engine#1823]
### TL;DR

Upgrading from v21 to v29 forces node pool recreation due to `gcfs_config` changing from being defined to a `dynamic` resource. This results in the plan outputting

```
          - gcfs_config { # forces replacement
              - enabled = false -> null
            }
```

### Expected behavior

I would expect given the impact of a node pool recreation that this particular change would be backwards compatible

### Observed behavior

```
  # module.gke.module.gke_cluster.module.gke.google_container_node_pool.pools["<pool name>"] must be replaced
-/+ resource "google_container_node_pool" "pools" {
...
      ~ node_config {
...
          - gcfs_config { # forces replacement
              - enabled = false -> null
            }
}
```

### Terraform Configuration

```hcl
node_pools = [
    {
      autoscaling       = true
      max_pods_per_node = 64
      name              = "<pool name>"
      machine_type      = "e2-highcpu-32"
      max_surge         = 1
      image_type        = "COS_CONTAINERD"
    }
  ]
```
```


### Terraform Version

```sh
v1.4.6
```

### Additional information

This is fixed in a local fork by replacing the following code in `modules/beta-private-cluster/cluster.tf`. The interesting thing is that this was implemented to fix a similar issue (https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1255). 

```
    dynamic "gcfs_config" {
      for_each = lookup(each.value, "enable_gcfs", false) ? [true] : []
      content {
        enabled = gcfs_config.value
      }
    }
```

with

```
    gcfs_config {
      enabled = var.enable_gcfs
    }
```

**Comment by davidgwps** [comment]
To follow up on this, I tried doing removing the node pool from the terraform state and reimporting it using v29 hoping that would set the state correctly but it still has the same plan output as above. If I hack the state file to set `gcfs_config` to null that also doesn't work and has the same plan output as above.

**Comment by onyx74** [comment]
HI @davidgwps,  recently have upgraded from 19v to 30. The rundom_id properties in **keepers** have been changed in the latest version of the module. To resolve node-pools recreation, you should manually include properties to terraform state file that were missing in the previous version of the module.

`
~ keepers = { +
...

"boot_disk_kms_key": "",
"enable_gcfs": "",
"enable_gvnic": "",
"enable_integrity_monitoring": "",
"enable_secure_boot": "",
"gpu_partition_size": "",
"max_pods_per_node": "",
"placement_policy": "",
"pod_range": ""`
